### ISSUE TITLE
Tasks for Spring Boot 2.7.0-GA 

### ISSUE THREAD

**Issue by Netyyyy** [Azure/azure-sdk-for-java#28726]
### Context

Spring Boot will release 2.7.0 GA on 2022-05-19.

We should upgrade external dependencies' version align with spring-boot-dependencies:2.7.0 managed versions. 

### Goal

1. Make our libraries compatible with latest Spring Boot: 2.7.0 by making external dependencies' version align to spring-boot-dependencies:2.7.0 managed versions

### Solution

Tasks are copied from this page: https://github.com/Azure/azure-sdk-for-java/wiki/Spring-Cloud-Azure-tasks-for-a-new-Spring-Boot-release

- Before azure-sdk-for-java in band release
  - [ ] Create a new issue to track the following tasks. Sample: [#28498](https://github.com/Azure/azure-sdk-for-java/issues/28498). And add the new created issue into [#28518](https://github.com/Azure/azure-sdk-for-java/issues/28518).
  - [ ] Read spring boot release notes. Sample: [Spring-Boot-2.6-Release-Notes](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-2.6-Release-Notes). Write down items we(Spring Cloud Azure team members) should care about. Just write these items by adding a new comment in the GitHub issue created in previous step.
  - [ ] Upgrade external dependencies' version according to spring-boot-dependencies and spring-cloud-dependencies. Refer to [related readme](https://github.com/Azure/azure-sdk-for-java/blob/aec4c6247ba7ba4de57dd866e3f5511ca4fbd387/sdk/spring/scripts/README.md). **NOTE**: This step should be finished before `Core code complete`. You can get more information about `Core code complete` in the Email with subject like `APRIL Release Kickoff`. 

- After azure-sdk-for-java in band release.

  Update spring-boot and spring-cloud version in these places:

  - [ ] [azure-spring-boot-samples](https://github.com/Azure-Samples/azure-spring-boot-samples)
  - [ ] [Reference doc](https://github.com/microsoft/spring-cloud-azure/tree/docs/docs/src/main/asciidoc).
  - [ ] [MS docs](https://docs.microsoft.com/en-us/azure/developer/java/spring-framework/).
  - [ ] [Wiki page: Spring Cloud Azure Timeline](https://github.com/Azure/azure-sdk-for-java/wiki/Spring-Cloud-Azure-Timeline)
  - [ ] [Wiki page: Spring Versions Mapping](https://github.com/Azure/azure-sdk-for-java/wiki/Spring-Versions-Mapping)
  - [ ] [Wiki page: Spring Support Multiple Versions](https://github.com/Azure/azure-sdk-for-java/wiki/Spring-Support-Multiple-Versions)

**Comment by andersthorbeck** [comment]
Hello,

My team's app's automatic update to Spring Boot 2.7.0 has been blocked by `com.azure.spring:spring-cloud-azure-dependencies:4.1.0`, claiming that 
>Spring Boot [2.7.0] is not compatible with this Spring Cloud Azure release
> Change Spring Boot version to one of the following versions [2.5.x, 2.6.x].

The full error was
```
org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.azure.spring.cloud.autoconfigure.implementation.compatibility.AzureSpringBootVersionVerifier]: Factory method 'springCloudAzureSpringBootVersionVerifier' threw exception; nested exception is com.azure.spring.cloud.autoconfigure.implementation.compatibility.AzureCompatibilityNotMetException: Spring Cloud Azure/ Spring Boot version compatibility checks have failed: [[VerificationResult@16e1219f description = 'Spring Boot [2.7.0] is not compatible with this Spring Cloud Azure release', action = 'Change Spring Boot version to one of the following versions [2.5.x, 2.6.x].
You can find the latest Spring Boot versions here [https://spring.io/projects/spring-boot#learn]. 
If you want to learn more about the Spring Cloud Azure compatibility, you can visit this page [https://github.com/Azure/azure-sdk-for-java/wiki/Spring-Versions-Mapping] and check the [Which Version of Spring Cloud Azure Should I Use] section.
If you want to disable this check, just set the property [spring.cloud.azure.compatibility-verifier.enabled=false]']]
```

This seems related to this GitHub issue. Do you have an estimated time for when Spring Cloud Azure will support Spring Boot 2.7.0?

**Comment by saragluna** [comment]
Hi @andersthorbeck, according to our compatibility test, our code is compatible with Spring Boot 2.7.0. So you can either disable the verifier by adding `spring.cloud.azure.compatibility-verifier.enabled=false` or specifying the support of 2.7.0 by adding `spring.cloud.azure.compatibility-verifier.compatible-boot-versions=2.7.0`. Our release to support Spring Boot 2.7.0 will be out next month (this is an estimation time).

**Comment by dbelyaev** [comment]
@saragluna I suppose, those property overrides should be taken with a note: "at your own risk", or not?

**Comment by saragluna** [comment]
@dbelyaev, we did run a compatibility test with Spring Boot 2.7.0, and it passed. But please let us know if you run into issues so that we can improve the compatibility tests.

**Comment by Netyyyy** [comment]
close it and works on [2.7.1](https://github.com/Azure/azure-sdk-for-java/issues/29658)
